src/mscorlib/src/System/Threading/ThreadPool.cs

```diff
@@ -1625,7 +1640,16 @@ private static bool QueueUserWorkItemHelper(WaitCallback callBack, Object state,
                 try { }
                 finally
                 {
-                    QueueUserWorkItemCallback tpcallBack = new QueueUserWorkItemCallback(callBack, state, compressStack, ref stackMark);
+                    ExecutionContext context = compressStack && !ExecutionContext.IsFlowSuppressed() ?
+                                                ExecutionContext.Capture(ref stackMark, ExecutionContext.CaptureOptions.IgnoreSyncCtx | ExecutionContext.CaptureOptions.OptimizeDefaultCase) :
+                                                ExecutionContext.Default;
```

Comment:
@ericeil should comment on whether for this version of the code base it needs to be this complicated, or whether for CoreCLR it could be simplified, since we know that many of these things are nops.